sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDDSuite.scala

```diff
@@ -49,8 +49,11 @@ class StateStoreRDDSuite extends SparkFunSuite with BeforeAndAfter with BeforeAn
   }
 
   override def afterAll(): Unit = {
-    super.afterAll()
-    Utils.deleteRecursively(new File(tempDir))
+    try {
+      Utils.deleteRecursively(new File(tempDir))
+    } finally {
+      super.afterAll()
```

Comment:
Could you check the history of this too? Is there a reason for this `StateStoreRDDSuite` to put `deleteRecursively` at the end?

Reply:
I cannot find other state store suite. Could you please let me know the name of these suites?

I intentionally changed the order since I saw this order (`deleteRecursively() ->super.afterAll()`) in the following other places.
Since I am neutral on this order, it is ok to keep the original order like
```
try {
  super.afterAll()
} finally {
  Utils.deleteRecursively(new File(tempDir))
}
```

```
core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala
core/src/test/scala/org/apache/spark/storage/DiskBlockManagerSuite.scala
sql/core/src/test/scala/org/apache/spark/sql/sources/CreateTableAsSelectSuite.scala
sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala
mllib/src/test/scala/org/apache/spark/ml/source/libsvm/LibSVMRelationSuite.scala
mllib/src/test/scala/org/apache/spark/ml/util/MLTest.scala
mllib/src/test/scala/org/apache/spark/ml/util/TempDirectory.scala
streaming/src/test/scala/org/apache/spark/streaming/rdd/MapWithStateRDDSuite.scala
```

Reply:
I prefer your change, too. But, the reason why I asked is that you didn't fix the order in [FlatMapGroupsWithStateSuite](https://github.com/apache/spark/pull/22337/files#diff-1e4641593c1f96a15e4502b1e53e721eR57) in this PR. I want to make it sure whether there is a obvious reason or not.

Reply:
Sure, let me change the order in `FlatMapGroupsWithStateSuite `, too.

Reply:
Hmm, I see. Let me keep the original order as 
```
try {
  super.afterAll()
} finally {
  Utils.deleteRecursively(...)
}
```
Thank you for listing them out.